assignment.rb

```diff
@@ -0,0 +1,85 @@
+class Yarn
+  attr_accessor :weight, :num_of_plies, :is_machine_wash, :yardage, :quantity
+
+  def total_yds
+    quantity * yardage
+  end
+
+  def use(type = "scarf")
+    puts "I will use this yarn to make a #{type}."
+  end
+
+  def rating
+    puts "Yarn is awesome!"
+  end
+end
+
+class Animal_Fiber < Yarn
+  attr_accessor :is_organic, :is_next_to_skin
+end
+
+class Plant_Fiber < Yarn
+  attr_accessor :is_organic
+end
+
+class Manmade_Fiber < Yarn
+  attr_accessor :colorway, :brand, :name
+
+  def rating
+    puts "Acrylic yarn sucks!"
+    super
+  end
+end
+
+class Wool_Yarn < Animal_Fiber
+  attr_accessor :colorway, :brand, :name
+
+  def use
+    super("sweater")
+  end
+end
+
+class Alpaca_Yarn < Animal_Fiber
+  attr_accessor :colorway, :brand, :name
+end
+
+class Cotton_Yarn < Plant_Fiber
+  attr_accessor :colorway, :brand, :name
```

Comment:
Given you've repeated these 3 attributes across numerous classes, I reckon you could store them all in one class and either add a `type` attribute or keep having them as sub-classes, just without these accessors being in every subclass.